### PR TITLE
Feat/maps for input data

### PR DESCRIPTION
**Pull request by joAschauer**
Hi @markushal, quite some changes, please have a look and consider this as a first proposal. Open tasks to discuss:

- [x] should the boxplots and maps of the input data be next to or above of each other?
- [x] do we still need the `st.radio` in the deep dive countries tab when we show the costs on top and the full load hours below?
- [x] do you want to keep the hovers as they are now or should we highlight the selected parameter that is shown on the map in the hover for "CAPEX" and "full load hours"
- [ ] can we do some refactoring in order to use the same reshaping and selection routines before `app.ptxboa_functions.edit_and_display_data_table()` and `app.ptxboa_functions._choropleth_map_world()` / `app.ptxboa_functions._choropleth_map_deep_dive_country()`

### CODE REVIEW

**Comment by markushal** [comment]
@joAschauer Great, thank you! 

> should the boxplots and maps of the input data be next to or above of each other?

next to each other. I think it still looks ok on a small laptop screen if the sidebar is not too wide. Can we somehow ensure that the height of the boxplot and the map are identical? That would be an improvement.

> do we still need the st.radio in the deep dive countries tab when we show the costs on top and the full load hours below?

no, that can be removed. 

> do you want to keep the hovers as they are now or should we highlight the selected parameter that is shown on the map in the hover for "CAPEX" and "full load hours"

highlighting it, or, maybe even better, adding a ""(displayed on map)" string or something similar, sounds like a good idea

> can we do some refactoring in order to use the same reshaping and selection routines before app.ptxboa_functions.edit_and_display_data_table() and app.ptxboa_functions._choropleth_map_world() / app.ptxboa_functions._choropleth_map_deep_dive_country()

Yes, please! Can you do that?

File: app/tab_deep_dive_countries.py [hunk 1]
```diff
@@ -84,8 +84,24 @@ def content_deep_dive_countries(api: PtxboaAPI, res_costs: pd.DataFrame) -> None
...
+        else:
+            map_parameter = "interest rate"
+        fig = plot_input_data_on_map(
+            api=api,
+            data_type=data_selection,
+            color_col=map_parameter,
+            scope=ddc,
+        )
+        st.plotly_chart(fig, use_container_width=True)
+    with st.expander(f"**Region specific {data_selection} data**"):
```
Comment (markushal) [inline app/tab_deep_dive_countries.py:104]: Just use ``Data:`` as header of the expander and remove the markdown header below

File: app/tab_deep_dive_countries.py [hunk 2]
```diff
@@ -99,7 +115,7 @@ def content_deep_dive_countries(api: PtxboaAPI, res_costs: pd.DataFrame) -> None
             column_config=column_config,
             key_suffix="_ddc",
         )
-    with c1:
+    with c2:
         # create plot:
         st.markdown("**Figure:**")
```
Comment (markushal) [inline app/tab_deep_dive_countries.py:120]: Set title to ``Regional distribution:``

File: app/tab_deep_dive_countries.py [hunk 3]
```diff
@@ -84,8 +84,24 @@ def content_deep_dive_countries(api: PtxboaAPI, res_costs: pd.DataFrame) -> None
         x = None
         st.markdown("TODO: fix surplus countries in data table")
 
-    c1, c2 = st.columns(2, gap="medium")
-    with c2:
+    st.subheader("Full load hours of renewable generation")
+    c1, c2 = st.columns([2, 1], gap="large")
+    with c1:
+        st.markdown("**Map of input data values**")
```
Comment (markushal) [inline app/tab_deep_dive_countries.py:90]: Set title to ``Map:``

Comment (markushal) [reply app/tab_deep_dive_countries.py:90]: I would use the same titles I suggested here on the input data tab as well

**Review by markushal — CHANGES_REQUESTED** [review]

File: app/tab_deep_dive_countries.py (hunk 1 above)
Comment (joAschauer) [reply app/tab_deep_dive_countries.py:104]: okay, I thought this is more explicit but don't have a strong opinion.

File: app/tab_deep_dive_countries.py (hunk 3 above)
Comment (joAschauer) [reply app/tab_deep_dive_countries.py:90]: I adjusted the titles as suggested and removed the colons as I think a title should go without a colon.

**Review by markushal — CHANGES_REQUESTED** [review]
Looks fine! I noticed one thing, I am not sure if it is because of the changes in this PR:

When editing flh for deep dive countries, the changes are saved to session state, and they are shown in the table and the map, but not in the box plot:

grafik

**Review by markushal — APPROVED** [review]
perfect, now it looks ready for merging

**Pull request merged**